### PR TITLE
Add tray for improved menubar integration

### PR DESCRIPTION
**Pull request by colemanw**
This is to support the (soon-to-be-in-core) [KAM](https://github.com/aydun/uk.squiffle.kam) extension.

drupal8

### CODE REVIEW

File: civicrm.module [hunk 1]
```diff
@@ -86,11 +86,23 @@ function civicrm_toolbar() {
...
-            'class' => ['toolbar-item'],
           ],
         ],
       ],
     ];
+    // Add tray if we are on a CiviCRM screen
+    $path = ltrim(\Drupal::service('path.current')->getPath(), '/');
+    if (strpos($path, 'civicrm') === 0) {
+      $items['civicrm']['tray'] = [
+        '#heading' => t('Test tray'),
```
Comment (JoeMurray) [inline civicrm.module]: Should this be removed from production code?

**Comment by JoeMurray** [comment]
What is a tray in this context?

File: civicrm.module (hunk 1 above)
Comment (colemanw) [reply civicrm.module]: Removed

**Comment by colemanw** [comment]
@JoeMurray it makes the CiviCRM menubar act like a Drupal taskbar tray. See screencast.

**Comment by JoeMurray** [comment]
Okay, watching the screencast about 10 times I eventually noticed that the CiviCRM entry on the top Drupal menu changes into a CiviCRM icon beside a menu hamburger when the CiviCRM menu disappears because the window gets too narrow. Got it. Nice.

**Comment by JoeMurray** [comment]
@monishdeb could we try this for our D8 site with the menuing that is giving us such grief.

**Comment by dsnopek** [comment]
The result and the code to accomplish it look quite similar to Roundearth :-)

**Comment by colemanw** [comment]
Thanks @JoeMurray and @dsnopek. Your testing and feedback of the KAM extension is appreciated.
@monishdeb when you test this out please leave a comment on the KAM PR also.

**Comment by monishdeb** [comment]
Tested, working fine in clean D8 instance. Merging now.

**Pull request merged**